packages/amplication-server/src/core/project/project.resolver.ts

```diff
@@ -92,9 +103,75 @@ export class ProjectResolver {
   )
   @InjectContextValue(InjectableOriginParameter.UserId, "data.user.connect.id")
   async commit(@Args() args: CreateCommitArgs): Promise<Commit | null> {
+    const projectId = args.data.project.connect.id;
+
+    await this.validateProject(projectId);
+
     return this.projectService.commit(args);
   }
 
+  async validateProject(projectId: string): Promise<void> {
+    const project = await this.projectService.findUnique({
+      where: { id: projectId },
+    });
+    const workspace = await this.workspaceService.getWorkspace({
+      where: { id: project.workspaceId },
+    });
+    const subscription = await this.subscriptionService.getCurrentSubscription(
+      workspace.id
+    );
+
+    if (!subscription) {
+      const projects = await this.projectService.findProjects({
+        where: { workspace: { id: workspace.id } },
+      });
+
+      const getServicesPerProject = projects.map(async (project) => {
+        const resources = await this.resourceService.resources({
+          where: {
+            projectId: project.id,
+          },
+        });
+
+        const services = resources.filter(
+          (x) => x.resourceType === EnumResourceType.Service
+        );
+
+        return services;
+      });
```

Comment:
What do you think about taking this part to an outside function?

Reply:
Moved to resource service.